src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/GenericLookupSignature.cs

```diff
@@ -107,12 +107,15 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
             {
                 Debug.Assert(_methodArgument.Unboxing == false);
 
+                bool useInstantiatingStub = (_methodArgument.Method.IsSharedByGenericInstantiations &&
+                    (_methodArgument.Method.Signature.IsStatic || _methodArgument.Method.HasInstantiation));
```

Comment:
Does this check also need to include value types?

Or should this condition be just `RequiresInstArg`?

Reply:
I have reverted this part of the change which doesn't work anyway as the method in question is "runtime determined", not "canonical". According to David Wrighton's analysis it turned out the problem had a different cause, namely imprecise comparison of generic context keys that could in very complex cases end up replacing a generic lookup with a different one.